docs/superpowers/plans/2026-07-22-optimize-vector-paths.md

```
@@ -0,0 +1,435 @@
+catch (e) { console.log('ERROR: ' + e.message); process.exitCode = 1; }
+```
+
+- [ ] **Step 2: Run the self-test (must prove the checker has teeth)**
+
+Run via `ctx_execute` (shell): `node build/vecopt/harness.js selftest`
+Expected stdout: `SELFTEST OK` (asserts `compactNum` is value-preserving on the battery, that `segEqual` REJECTS a rounding transform, and that a real `optimize` round-trips). If it prints `ERROR: …`, fix the harness before continuing.
+
+- [ ] **Step 3: Run `check` on all three files**
+
+Run via `ctx_execute` (shell), one call:
+```
```

markdownlint-cli2 (0.23.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)